src/Compilers/Shared/BuildClient.cs

```diff
@@ -51,23 +54,23 @@ internal RunCompilationResult RunCompilation(IEnumerable<string> originalArgumen
 
             var args = originalArguments.Select(arg => arg.Trim()).ToArray();
 
-            bool hasShared;
-            string keepAlive;
-            string errorMessage;
-            string sessionKey;
-            List<string> parsedArgs;
             if (!CommandLineParser.TryParseClientArgs(
                     args,
-                    out parsedArgs,
-                    out hasShared,
-                    out keepAlive,
-                    out sessionKey,
-                    out errorMessage))
+                    out List<string> parsedArgs,
+                    out bool hasShared,
+                    out string keepAlive,
+                    out string sessionKey,
+                    out string errorMessage))
             {
                 textWriter.WriteLine(errorMessage);
                 return RunCompilationResult.Failed;
             }
 
+            if (!TryEnableMulticoreJitting(out errorMessage))
+            {
+                textWriter.WriteLine(errorMessage);
```

Comment:
>textWriter [](start = 16, length = 10)

It looks like the ```textWriter``` can be null. Why is it safe to call ```WriteLine``` of off it?

Reply:
I don't think this `textWriter` can ever be null. We check for null at the top of the file and, if it is, we set it to Console.Out, which can never be null.